packages-web/maps/src/components/LeafletMap.tsx

```diff
@@ -1,167 +1,101 @@
 }
 
-function addMarkers(
-    map: Map,
-    markerGroup: FeatureGroup,
-    locations: Marker[],
-    currentLocation: Option<Marker>,
-    zoomLevel: number,
-    autoZoom: boolean
-): void {
-    markerGroup.clearLayers();
-    const defaultCenterLocation = { lat: 51.906688, lng: 4.48837 };
-
-    locations
-        .concat(currentLocation ? [currentLocation] : [])
-        .map(location => addMarker(location))
-        .filter(m => m)
-        .forEach(marker => markerGroup.addLayer(marker));
-
-    const bounds = markerGroup.getBounds();
-
-    if (bounds.isValid()) {
-        map.fitBounds(bounds, { animate: false }).invalidateSize();
-    }
-    if (!autoZoom) {
-        map.panTo(
-            {
-                lat: locations?.[0]?.latitude ?? defaultCenterLocation.lat,
-                lng: locations?.[0]?.longitude ?? defaultCenterLocation.lng
-            },
-            { animate: false }
-        );
-        map.setZoom(zoomLevel);
-    }
-}
-
-function addMarker(marker: Marker): LeafletMarker {
-    const mapMarker = new LeafletMarker([Number(marker.latitude), Number(marker.longitude)], {
-        title: marker.title,
-        alt: marker.title
-    });
-    if (marker.title) {
-        mapMarker.setTooltipContent(marker.title);
-        if (marker.onClick) {
-            const infoContent = document.createElement("span");
-            infoContent.innerHTML = marker.title || "";
-            infoContent.style.cursor = "pointer";
-            infoContent.onclick = marker.onClick;
-            mapMarker.bindPopup(infoContent).openPopup();
-        } else {
-            mapMarker.bindPopup(marker.title).openPopup();
-        }
-    } else {
-        if (marker.onClick) {
-            mapMarker.on("click", () => {
-                marker.onClick!();
-            });
-        }
-    }
-    if (marker.url) {
-        mapMarker.setIcon(
-            icon({
-                iconUrl: marker.url,
-                iconSize: [32, 32],
-                className: "marker"
-            })
-        );
-    }
-    return mapMarker;
-}
-
-function baseMapLayer(mapProvider: MapProviderEnum, mapsToken?: string): TileLayer {
-    let urlTemplate;
-    let mapAttribution;
-    if (mapProvider === "mapBox") {
-        urlTemplate = customUrls.mapbox + mapsToken;
-        mapAttribution = mapAttr.mapboxAttr;
-    } else if (mapProvider === "hereMaps" && mapsToken && mapsToken.indexOf(",") > 0) {
-        const splitToken = mapsToken.split(",");
-        urlTemplate = customUrls.hereMaps + `app_id=${splitToken[0]}&app_code=${splitToken[1]}`;
-        mapAttribution = mapAttr.hereMapsAttr;
-    } else {
-        urlTemplate = customUrls.openStreetMap;
-        mapAttribution = mapAttr.openStreetMapAttr;
-    }
-
-    return tileLayer(urlTemplate, {
-        attribution: mapAttribution,
-        id: mapProvider === "mapBox" ? "mapbox.streets" : undefined
-    } as TileLayerOptions);
+function LeafletMarker({ marker }: { marker: Marker }): ReactElement {
```

Comment:
🎄  Here there is no big need for this component. And if you just inline its logic into `LeafletMap` you can increase the coverage of your snapshots 😉